### ISSUE TITLE
ReceiveAsync looses Context after await / SynchronizedDispatcher (?)

### ISSUE THREAD

**Issue by minesworld** [akkadotnet/akka.net#6961]
**Version Information**
1.5.13

**Describe the bug**
after await of an async call Akka.net throws a bug on .Tell()

**To Reproduce**
1. Download VS2022 solution from https://github.com/minesworld/AkkaWebView2Test
2. Build & Run the WebView2Test project - this is the "non Akka" implementation
3. Build & Run the AkkaWebView2Test project, 
4. the code in ReceiveAsync does two `Sender.Tell(message)` - before and **after** the await async
5. See error on Debug output: 
```
Exception thrown: 'System.NotSupportedException' in Akka.dll
after: System.NotSupportedException: There is no active ActorContext, this is most likely due to use of async operations from within this actor.
   at Akka.Actor.ActorBase.get_Context()
   at Akka.Actor.ActorBase.get_Sender()
   at AkkaWebView2Test.WebView2Actor.<.ctor>b__0_0(Work message)
```

**Expected behavior**
Should "just" work...

**Environment**
Windows 11 22H2 22621.2428
AppSDK 1.4

**Additional context**
I don't know if this is due to the usage of WebView2 and/or the SynchronizedDispatcher ...

**Comment by minesworld** [comment]
Seems to be the workaround is to store the needed properties as Sender in a variable before the async call.

See the WorkAround project ... 

**If that is the solution - would be nice to have it in the documentation.**

**Not Working:** 

```
ReceiveAsync<Work>(async message =>
{
    try
    {
        Sender.Tell(message);
    }
    catch(Exception ex)
    {
        System.Diagnostics.Debug.WriteLine($"before: {ex}");
    }

    await message.WebView.EnsureCoreWebView2Async();
    message.Window.Content = message.WebView;
    message.WebView.CoreWebView2.Navigate("https://www.microsoft.com/surface");

    try
    {
        Sender.Tell(message);
    }
    catch (Exception ex)
    {
        System.Diagnostics.Debug.WriteLine($"after: {ex}");
    }
});
```

**Working:** 

```
ReceiveAsync<Work>(async message =>
{
    var sender = Sender; // seems to solve the problem... hmmm ???

    try
    {
        sender.Tell(message);
    }
    catch (Exception ex)
    {
        Debug.WriteLine($"before: {ex}");
    }

    await message.WebView.EnsureCoreWebView2Async();
    message.Window.Content = message.WebView;
    message.WebView.CoreWebView2.Navigate("https://www.microsoft.com/surface");

    try
    {
        sender.Tell(message);
    }
    catch (Exception ex)
    {
        Debug.WriteLine($"after: {ex}");
    }
});
```

But I will try the work around in a more complex scenario before closing the bug and filing a new issue regarding the documentation... . Just to be sure.

**Comment by to11mtm** [comment]
> `        sender.Tell(message);`

Technically, this can also be a hazard and the implicit `Self` could be incorrect (e.x. `NoSender` instead of the expected Actor).

This is a known issue, #6279 probably has the most discussion around it.

That said, I've been thinking about this and wonder whether there are some less heavy-lift ways to fix some of these issues...

**Comment by Aaronontheweb** [comment]
This is a duplicate of #6279 so we'll close this issue, but thanks for filing an organized ticket - we really appreciate it. We are eager to fix this in v1.6.
